src/components/internal/Popover.tsx

```diff
@@ -25,7 +25,7 @@ export function Popover(props: PopoverProps) {
 
         // If we clicked the trigger element (or within it), then that will call a `state.toggle` for us.
         // Return early if that happens, otherwise we'd call `onClose`, then the trigger would toggle it back open.
-        if (triggerRef.current?.contains(e)) {
+        if (triggerRef.current?.contains(e) || e.closest('[data-overlay-container]')) {
```

Comment:
**question:** Why do we want to return `true` when a user interacts with the `data-overlay-container`?

Reply:
I ask because I think when we return `true` in `shouldCloseOnInteractOutside`, then we are saying "Yes, close this overlay". But in this instance, when we interact with other elements within the Popover, we don't really want it to close. I also think that we should copy the logic used in the Modal component's `shouldCloseOnInteractOutside` method.